### PR TITLE
Added types for klaw

### PR DESCRIPTION
**Pull request by mceachen**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

Thanks for DefinitelyTyped!

### CODE REVIEW

File: klaw/klaw.d.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Type definitions for klaw
```
Comment (andy-ms) [inline klaw/klaw.d.ts]: Please include a semver at the end of this line

File: klaw/klaw.d.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+// Project: https://github.com/jprichardson/node-klaw
+// Definitions by: Matthew McEachen <https://github.com/mceachen>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../node/node.d.ts" />
+
+declare module "klaw" {
+
+  import * as fs from "fs"
```
Comment (andy-ms) [inline klaw/klaw.d.ts]: Use 4 spaces or 1 tab to indent.

File: klaw/klaw.d.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    interface Item {
+      path: string
+      stats: fs.Stats
+    }
+
+    type QueueMethod = "shift" | "pop"
+
+    interface Options extends ReadableOptions {
+      queueMethod?: QueueMethod
+      pathSorter?: (a: Array<Item>) => Array<Item>
```
Comment (andy-ms) [inline klaw/klaw.d.ts]: Prefer `Item[]`

**Comment by mceachen** [comment]
@andy-ms thanks for the review.

**Comment by mceachen** [comment]
@andy-ms when will the npm get published? I looked around the docs (like [this](http://definitelytyped.org/guides/pull-request.html)) and didn't see if I needed to do anything else to make `@types/klaw` a thing.

Thanks.

**Pull request merged**